[MSE][GStreamer] update maximum buffer size after receiving first init segment

eocanha opened the pull request:
#### da13f3fd899ebd06434295073c30bbfd9dd34241
<pre>
[MSE][GStreamer] update maximum buffer size after receiving first init segment
<a href="https://bugs.webkit.org/show_bug.cgi?id=296059">https://bugs.webkit.org/show_bug.cgi?id=296059</a>

Reviewed by Xabier Rodriguez-Calvar.

The current
SourceBuffer::sourceBufferPrivateDidReceiveInitializationSegment() code
only calls SourceBufferPrivate::setMaximumBufferSize() on new init
segments when the segment has (at least) a video track. Maximum buffer
size is managed with high granularity in WPE by using the
MSE_MAX_BUFFER_SIZE environment variable. For that reason, that call
must happen independently of the kind of tracks present (at least on the
WPE platform). This will ensure that the proper frame eviction code in
SourceBufferPrivate::setMaximumBufferSize() is called in all cases.

See: <a href="https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1534">https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1534</a>

This patch calls SourceBufferPrivate::setMaximumBufferSize()
inconditionally on the WPE platform.

Original author: Eugene Mutavchi &lt;Ievgen_Mutavchi@comcast.com&gt;

* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
(WebCore::SourceBuffer::sourceBufferPrivateDidReceiveInitializationSegment): Disable the if, so the call happens inconditionally on WPE.

Canonical link: <a href="https://commits.webkit.org/297674@main">https://commits.webkit.org/297674@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/757f9bb36f6781c95f8d186bdc04a85cd442370b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/112523 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/32255 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/22733 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/118721 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/62973 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/114485 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/32907 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/40818 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/85638 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/36248 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/115470 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/26260 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/101229 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/65940 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25558 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/19363 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/62481 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/95650 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/19436 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/121943 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/39597 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29488 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/94504 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/39979 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/97462 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94244 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24056 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/39357 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17160 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/35677 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/39485 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/44973 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/39122 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/42457 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/40863 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

calvaris reviewed (approved):
There's a changelog line that is too long, please shorten it.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/757f9bb36f6781c95f8d186bdc04a85cd442370b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48121

eocanha committed:
[MSE][GStreamer] update maximum buffer size after receiving first init segment
https://bugs.webkit.org/show_bug.cgi?id=296059

Reviewed by Xabier Rodriguez-Calvar.

The current
SourceBuffer::sourceBufferPrivateDidReceiveInitializationSegment() code
only calls SourceBufferPrivate::setMaximumBufferSize() on new init
segments when the segment has (at least) a video track. Maximum buffer
size is managed with high granularity in WPE by using the
MSE_MAX_BUFFER_SIZE environment variable. For that reason, that call
must happen independently of the kind of tracks present (at least on the
WPE platform). This will ensure that the proper frame eviction code in
SourceBufferPrivate::setMaximumBufferSize() is called in all cases.

See: https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1534

This patch calls SourceBufferPrivate::setMaximumBufferSize()
inconditionally on the WPE platform.

Original author: Eugene Mutavchi <Ievgen_Mutavchi@comcast.com>

* Source/WebCore/Modules/mediasource/SourceBuffer.cpp:
(WebCore::SourceBuffer::sourceBufferPrivateDidReceiveInitializationSegment): Disable the if, so the call happens inconditionally on WPE.

Canonical link: https://commits.webkit.org/297674@main
